projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/GraphQLDataFetchers.java

```diff
@@ -207,8 +215,9 @@ private Predicate<V2DataJob> computeSearch(
   private List<V2DataJob> populateDeployments(
       List<V2DataJob> allDataJob, Map<String, DataJob> dataJobs) {
     Map<String, JobDeploymentStatus> deploymentStatuses =
-        deploymentService.readDeployments().stream()
-            .collect(Collectors.toMap(JobDeploymentStatus::getDataJobName, cronJob -> cronJob));
+        (dataJobDeploymentPropertiesConfig.getReadDataSource().equals(ReadFrom.DB))
```

Comment:
Add an explicit check for K8S or else empty list.

Reply:
Not sure I understand. There is additional check necessary like
```java
if (dataJobDeploymentPropertiesConfig.getReadDataSource().equals(ReadFrom.DB)) {
    Map<String, JobDeploymentStatus> deploymentStatuses = readJobDeploymentsFromDb();
}
else if (dataJobDeploymentPropertiesConfig.getReadDataSource().equals(ReadFrom.K8S)) {
    Map<String, JobDeploymentStatus> deploymentStatuses = readJobDeploymentsFromK8s();
}
else {
    Map<String, JobDeploymentStatus> deploymentStatuses = Collections.emptyMap();
}
```
or am I misunderstanding?

Reply:
correct